### PR TITLE
Moved golangci linter config to the separate file

### PR DESCRIPTION
**Pull request by begmaroman**
# Description

Moved golangci config to the single file so github actions and local setup could use the same config file. Fixed existing linter issues. 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

**Comment by Kourin1996** [comment]
Looks good. Are you going to resolve all lint errors caused by new linter configuration in this PR?

**Review by Kourin1996 — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]
This will play nicely with our local IDE, nice one!

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  skip-dirs-use-default: true
+
+service:
+  golangci-lint-version: 1.49.0
+
+linters:
+  disable-all: true
+  enable:
+    - whitespace # Tool for detection of leading and trailing whitespace
+    - wsl # Whitespace Linter - Forces you to use empty lines!
```
Comment (zivkovicmilos) [inline .golangci.yml]: No need to use exclamation marks for comments

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    - wastedassign # Finds wasted assignment statements.
```
Comment (zivkovicmilos) [inline .golangci.yml]: No need for the period at the end of the comment, other comment lines in the same file have no period

File: .golangci.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    - unconvert # Unnecessary type conversions
+    - tparallel # detects inappropriate usage of t.Parallel() method in your Go test codes
```
Comment (zivkovicmilos) [inline .golangci.yml:20]: Comment begins with a lowercase letter, when it should begin with an uppercase one

File: .golangci.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - thelper # Detects golang test helpers without t.Helper() call and checks the consistency of test helpers
+    - stylecheck # Stylecheck is a replacement for golint
+    - prealloc # Finds slice declarations that could potentially be pre-allocated
+    - predeclared # Find code that shadows one of Go's predeclared identifiers
```
Comment (zivkovicmilos) [inline .golangci.yml]: Should be `Finds` instead of `Find`, the other comments are formed in the active verb voice

File: .golangci.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    - lll # Long lines
+    - importas # Enforces consistent import aliases
+    - gosec # Security problems
+    - gofmt # Whether the code was gofmt-ed
+    - goimports # Unused imports
+    - goconst # Repeated strings that could be replaced by a constant
+    - forcetypeassert # Finds forced type assertions
+    - dogsled # Checks assignments with too many blank identifiers (e.g. x, , , _, := f())
+    - dupl # Code clone detection
+    - errname # Checks that sentinel errors are prefixed with the Err and error types are suffixed with the Error.
```
Comment (zivkovicmilos) [inline .golangci.yml]: Same as the previous comment, comment should not end in a period

File: .golangci.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    - errorlint # errorlint is a linter for that can be used to find code that will cause problems with the error wrapping scheme introduced in Go 1.13.
```
Comment (zivkovicmilos) [inline .golangci.yml]: Same as the previous comments, comment should not end in a period, and not start with a lowercase letter

File: .golangci.yml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    - path: _test\.go
+      linters:
+        - gosec
+        - unparam
+        - lll
+  include:
+    - EXC0012 # EXC0012 revive: Annoying issue about not having a comment. The rare codebase has such comments
+    - EXC0013 # EXC0013 revive: Annoying issue about not having a comment. The rare codebase has such comments
+    - EXC0014 # EXC0014 revive: Annoying issue about not having a comment. The rare codebase has such comments
+    - EXC0015 # EXC0015 revive: Annoying issue about not having a comment. The rare codebase has such comments
```
Comment (zivkovicmilos) [inline .golangci.yml]: I'd update the comments here to just have the pattern, not the `why` portion, as they're all the same:
https://golangci-lint.run/usage/false-positives/#exc0012

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

I've left a couple of comments on the config file itself

**Review by 0xAleksaOpacic — APPROVED** [review]
LGTM

File: .golangci.yml (hunk 7 above)
Comment (begmaroman) [reply .golangci.yml]: Done

File: .golangci.yml (hunk 6 above)
Comment (begmaroman) [reply .golangci.yml]: Fixed

File: .golangci.yml (hunk 5 above)
Comment (begmaroman) [reply .golangci.yml]: Fixed

File: .golangci.yml (hunk 4 above)
Comment (begmaroman) [reply .golangci.yml]: Fixed

File: .golangci.yml (hunk 3 above)
Comment (begmaroman) [reply .golangci.yml:20]: Fixed

File: .golangci.yml (hunk 2 above)
Comment (begmaroman) [reply .golangci.yml]: Removed

File: .golangci.yml (hunk 1 above)
Comment (begmaroman) [reply .golangci.yml]: Removed

**Pull request merged**